app/customMetricsClient.js

```diff
@@ -0,0 +1,424 @@
+/**
+ * Copyright 2023 IBM Corp. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+const { Counter, Gauge, Histogram } = require('prom-client');
+// --- Define Custom API Metrics ---
+
+// --- Other API Action Metrics ---
+// Count how many API calls occur
+const signInHistogram = new Histogram({
+  name: 'sign_in_duration_seconds',
+  help: 'Duration of signIn operations in seconds',
+  buckets: [0.1, 0.5, 1, 2, 5]
+});
+
+const apiCallsCounter = new Counter({
+  name: 'my_api_calls_total',
+  help: 'Total number of API calls'
+});
+
+// Gauge success and failure of signIn
+const signInGauge = new Gauge({
+  name: 'sign_in_result_status',
+  help: 'Total number of signIn operations, labeled by success or failure',
+  labelNames: ['status'],
+});
+
+// --- Cluster Resolver API Metrics ---
+// Count duration of clustersByClusterId
+const clusterByClusterIdHistogram = new Histogram({
```

Comment:
I think for both of the histogram and the gauge metrics the operation can be a label on the metric so you can consolidate all of these down into a single metric definition. Where the labels indicate what the action is

Reply:
I will take a look and see if I can consolidate, thanks!